tests/integration/common/s2n_test_common.py

```diff
@@ -87,6 +87,10 @@ def __test(scenario):
         server = None
         try:
             server, client = connect(get_peer, scenario)
+            if client.poll():
+                raise AssertionError("Client process crashed")
+            if server.poll():
+                raise AssertionError("Server process crashed")
             if test_func:
```

Comment:
You might need to check after test_func is called, or we could miss a crash.

Reply:
who uses test_func() at this time?